.circleci/publish_to_github.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# TODO(josephperrott): Remove this script in favor of common orb to perform the task.
+
+# Script to publish the build artifacts to a GitHub repository.
+# Builds will be automatically published once new changes are made to the repository.
+
+# The script should immediately exit if any command in the script fails.
+set -e
+
+# Go to the project root directory
+cd $(dirname ${0})/..
+
+if [ -z ${DEV_INFRA_BUILDS_TOKEN} ]; then
+  echo "Error: No access token for GitHub could be found." \
+       "Please set the environment variable 'DEV_INFRA_BUILDS_TOKEN'."
+  exit 1
+fi
+
+# Function to publish artifacts of a package to Github.
+publishPackage() {
+  packageName="dev-infra-private"
+  packageRepo="dev-infra-private-builds"
+  buildDir="$(pwd)/dist/bin/npm_package"
+  buildVersion=$(node -pe "require('./package.json').version")
+  branchName="master"
```

Comment:
You might want to keep the `CIRCLE_BRANCH` name like in the COMP repo (in case we ever want to deploy snapshots for e.g. a development branch; we can then control the deployment in the Circle config)